packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -81,34 +79,35 @@ function PanZoomMesh() {
       // Prevent events from reaching tooltip mesh when panning
       evt.stopPropagation();
 
-      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
-      const { x: pointerX, y: pointerY } = projectedPoint;
-      const { x: startX, y: startY } = startOffsetPosition.current;
+      const delta = startOffsetPosition.current
+        .clone()
+        .sub(evt.unprojectedPoint);
+      const target = camera.position.clone().add(delta);
 
-      const targetX = startX - pointerX;
-      const targetY = startY - pointerY;
-
-      moveCameraTo(targetX, targetY);
+      moveCameraTo(target.x, target.y);
     },
     [camera, moveCameraTo]
   );
 
   const onWheel = useCallback(
     (evt: ThreeEvent<WheelEvent>) => {
-      const { sourceEvent } = evt;
+      const { sourceEvent, unprojectedPoint } = evt;
       const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
 
-      camera.zoom = Math.max(1, camera.zoom * factor);
-      camera.updateProjectionMatrix();
+      const zoomVector = new Vector3(1 / factor, 1 / factor, 1);
+      camera.scale.multiply(zoomVector).clamp(ZERO_VECTOR, ONE_VECTOR);
```

Comment:
We can't really go below 0, though. Could we just clamp to `ONE_VECTOR` with `min()` or something?

Reply:
Good call